Servers/Administration/OJS.Servers.Administration/Controllers/ProblemsController.cs

```diff
@@ -607,17 +645,41 @@ private static void AddSubmissionTypes(Problem problem, AdminActionContext actio
     {
         var newSubmissionTypes = actionContext.GetSubmissionTypes()
             .Where(x => x.IsChecked)
-            .Select(x => new SubmissionTypeInProblem
+            .Select(x =>
             {
-                ProblemId = problem.Id,
-                SubmissionTypeId = int.Parse(x.Value!.ToString() !),
-                SolutionSkeleton = x.Expand != null && x.Expand.Value != null
-                    ? x.Expand.Value!.ToString() !.Compress()
-                    : Array.Empty<byte>(),
+                var psted = new ProblemSubmissionTypeExecutionDetails();
+                psted.ProblemId = problem.Id;
+                psted.SubmissionTypeId = int.Parse(x.Value!.ToString() !);
+                if (x.Expand == null)
+                {
+                    return psted;
+                }
+
+                psted.SolutionSkeleton = x.Expand.First(y =>
+                    y.Name == x.Name + " " + AdditionalFormFields.SolutionSkeletonRaw).Value != null
+                    ? x.Expand.First(y => y.Name == x.Name + " " + AdditionalFormFields.SolutionSkeletonRaw)
+                        .Value!.ToString() !.Compress()
+                    : Array.Empty<byte>();
+
+                var timeLimitValue = x.Expand.First(y =>
+                    y.Name == x.Name + " " + AdditionalFormFields.TimeLimit).Value;
+                if (timeLimitValue != null)
+                {
+                    psted.TimeLimit = string.IsNullOrEmpty(timeLimitValue.ToString()) ? null : int.Parse(timeLimitValue.ToString() !);
+                }
+
+                var memoryLimitValue = x.Expand.First(y =>
+                    y.Name == x.Name + " " + AdditionalFormFields.MemoryLimit).Value;
+                if (memoryLimitValue != null)
+                {
+                    psted.MemoryLimit = string.IsNullOrEmpty(memoryLimitValue.ToString()) ? null : int.Parse(memoryLimitValue.ToString() !);
+                }
```

Comment:
Why not just use the check `string.IsNullOrEmpty(memoryLimitValue.ToString())` directly in the `if`? For me this is somehow repeating the same check twice.

Reply:
I need to check if the memoryLimitValue is null first, otherwise it throws an error when I call toString() on it.

Reply:
You can use `memoryLimitValue?.ToString()`

Reply:
Done.